torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -1320,13 +1320,31 @@ struct to_ir {
     pushFrame(comprehension_block);
     WithInsertPoint guard(comprehension_block);
     auto emit_body = [&]() {
-      auto comprehension_out = emitExpr(lc.elt());
+      Value* out = emitExpr(lc.elt());
+
+      // Make sure that any element types are subtypes of the annotated
+      // type
+      std::stringstream ss;
+      if (type_hint && !out->type()->isSubtypeOfExt(type_hint->expect<ListType>()->getElementType(), &ss)) {
+        throw ErrorReport(loc) << ss.str() << "List type annotation did"
```

Comment:
`ss` contains the reason for not satisfying subtype relationship, it should ideally be at the end of error message.

Reply:
thanks!